File: java/src/com/chanse/code/interfaces/CalculatorTest.java
package com.chanse.code.interfaces;

public class CalculatorTest {

    public static void main(String[] args) {
        var calc = new CalculatorImpl();
        calc.add(2, 3);
        calc.sub(2, 3);
        Calculator.mul(2, 3);
    }
}

File: kotlin/src/com/chanse/code/interfaces/CalculatorTest.kt
package com.chanse.code.interfaces

fun main() {
    val calc = DefaultCalculator()
    calc.add(2, 3)
    calc.sub(2, 3)

    val printingCalc = PrintingCalculator()
    printingCalc.add(2, 3)
    printingCalc.sub(2, 3)
}
